openmdao/core/driver.py

```diff
@@ -398,19 +398,20 @@ def set_desvar(self, name, value):
         val : ndarray or float
             value to set the parameter
         """
-        if self.root.unknowns._dat[name].val.size == 0:
+        val = self.root.unknowns._dat[name].val
+        if not isinstance(val, _ByObjWrapper) and \
+                       self.root.unknowns._dat[name].val.size == 0:
```

Comment:
We talked about defining something like driver.supports['uses_gradients'] so that we can allow pbo desvars on gradient-free drivers, but not on gradient drivers.